lib/status-bar-manager.coffee

```diff
@@ -14,9 +19,10 @@ class StatusBarManager
 
   initialize: (@statusBar) ->
 
-  update: (currentMode) ->
+  update: (currentMode, currentSubmode) ->
+    currentFullMode = currentMode + "$" + currentSubmode if currentSubmode?
```

Comment:
You're right, I missed that. OK, enumeration it is; the longer I've thought about it the better this option has sounded anyway. Commit coming soon.